### DISCUSSION TITLE
Lesson 6 - error sending request for url (http://127.0.0.1:7545/)

### DISCUSSION THREAD

**Discussion by Perso77** [Cyfrin/foundry-full-course-cu#37] [Q&A] [answered]
i try to connect with ganache RPC server but still not working !! 
![deploy](https://github.com/Cyfrin/foundry-full-course-f23/assets/116321001/d6ecd62d-b369-4835-b4fe-fc81c81b7b25)

**Comment by cromewar** [comment, marked as answer]
Hello there @Perso77 if you are running this on WSL, be sure to change the host on ganache to WSL and use the new provided IP.

**Reply by Perso77** [reply]
can you explain me how to do this?

**Reply by cromewar** [reply]
Sure, open Ganache UI, then open settings and under hostname instead of selecting the default localhost, select the one which says WSL:

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/20199180/a22cc090-1292-464e-8f10-687b9b6f9bbf)

**Reply by PatrickAlphaC** [reply]
@cromewar can you add this to the `chronological_updates.md` file?

**Reply by cromewar** [reply]
Yes Sir! @PatrickAlphaC

**Reply by AgniveshChaubey** [reply]
any idea how to change the IP for anvil?

**Reply by AgniveshChaubey** [reply]
I am getting error in Chain ID here 
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/113545196/d58e5152-2a4c-4bbf-8d21-b3801775a52d)

**Reply by alymurtazamemon** [reply]
@AgniveshChaubey Try `1337`

**Reply by AgniveshChaubey** [reply]
`1337` and `31337` both are not working

**Reply by alymurtazamemon** [reply]
@AgniveshChaubey 

<img width="719" alt="Screenshot 2023-06-14 at 12 59 36 PM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/56123405/0825f050-d3b4-4898-a40c-d098ab05bb98">

**Reply by alymurtazamemon** [reply]
@AgniveshChaubey Search this error on google maybe some settings are off in your metamask.

**Reply by AgniveshChaubey** [reply]
Got the solution [here](https://book.getfoundry.sh/reference/anvil/#executor-environment-config)--
Just needed to set a new chain Id - `anvil --chain-id <your chain id>`
e.g - `anvil --chain-id 31337`

**Reply by alymurtazamemon** [reply]
@AgniveshChaubey Ah! thanks for this question, I found the default id is `1` in foundry [link](https://github.com/foundry-rs/foundry/tree/master/config#default-profile)

**Comment by Perso77** [comment]
thanks for help
![ganachee](https://github.com/Cyfrin/foundry-full-course-f23/assets/116321001/b2caebfd-5c04-4f0d-a6c0-4429b109fa57)
